player/brains/cPlayerBrainCampaign.cpp

```diff
@@ -1680,14 +1686,71 @@ namespace brains {
 
             }
         } else {
-            // non peaceful state, what to do? react? etc.
+            if (!wormsign) {
+                if (event.player == player) {
+                    // i discovered something
+                    if (event.entityType == eBuildType::UNIT) {
+                        cUnit &pUnit = unit[event.entityID];
+                        if (pUnit.isValid() && !pUnit.getPlayer()->isSameTeamAs(player)) {
+                            if (m_centerOfBaseCell > -1 && map.distance(m_centerOfBaseCell, event.atCell) < 20) {
```

Comment:
make `20` not a magic number but a variable. Some kind of 'search area' for units?